src/Todoist.Net/Serialization/Converters/BoolConverter.cs

```diff
@@ -1,28 +1,35 @@
-﻿using System;
-
-using Newtonsoft.Json;
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 namespace Todoist.Net.Serialization.Converters
 {
-    internal class BoolConverter : JsonConverter
+    internal class BoolConverter : JsonConverter<bool>
     {
-        public override bool CanConvert(Type objectType)
+        public override bool Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            return objectType == typeof(bool);
-        }
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.String:
+                    return reader.GetString() == "1";
 
-        public override object ReadJson(
-            JsonReader reader,
-            Type objectType,
-            object existingValue,
-            JsonSerializer serializer)
-        {
-            return reader.Value?.ToString() == "1";
+                case JsonTokenType.Number:
+                    return reader.GetInt32() == 1;
+
+                case JsonTokenType.True:
+                    return true;
+
+                case JsonTokenType.False:
+                    return false;
+
+                default:
+                    throw new NotImplementedException();
```

Comment:
I think we should use another exception type here. Like InvalidOperationException or ArgumentOutOfRange exception with the TokenType in the ctor or kind of JsonException. NotImplementedException means that we will implement something in the future.

Reply:
You are right, I missed this point.

Reply:
What about `JsonException`?

```c#
throw new JsonException($"Token type not supported for boolean properties. Token: {reader.TokenType}");
```

Reply:
Yep, let's use `JsonException`